theano/tensor/nnet/nnet.py

```diff
@@ -756,7 +757,8 @@ def local_logsoftmax(node):
         return [ret]
 
 
-@opt.register_specialize('stabilize', 'fast_compile')
+@opt.register_stabilize('fast_compile')
+@opt.register_specialize('fast_compile')
 @gof.local_optimizer([SoftmaxGrad])
 def local_logsoftmax_grad(node):
     """
```

Comment:
For some reason, this seems to have disrupted the giant cross-entropy optimization.